retrievalmarket/impl/lazyblockstore_test.go

```diff
@@ -0,0 +1,93 @@
+package retrievalimpl
+
+import (
+	"context"
+	"testing"
+
+	ds "github.com/ipfs/go-datastore"
+	bstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/dagstore"
+
+	"github.com/filecoin-project/go-fil-markets/shared_testutil"
+)
+
```

Comment:
Please can you also write a test to verify that the lazy blockstore loads the underlying blockstore only once ? Can pass a `CountingBlockstore` wrapper to the lazy blockstore...

Reply:
https://github.com/filecoin-project/go-fil-markets/pull/560/commits/296d14c97abc885ba740a271187aeafec8c3f16a